core/src/main/java/io/aiven/klaw/service/EnvsClustersTenantsControllerService.java

```diff
@@ -1346,11 +1347,14 @@ public EnvUpdatedStatus getUpdateEnvStatus(String envId) throws KlawException {
               env.getType(),
               kwClusters.getKafkaFlavor(),
               tenantId);
+
     } catch (Exception e) {
       status = ClusterStatus.OFFLINE;
       log.error("Error from getUpdateEnvStatus ", e);
     }
     env.setEnvStatus(status);
+    kwClusters.setClusterStatus(status);
+    manageDatabase.getHandleDbRequests().addNewCluster(kwClusters);
     manageDatabase.getHandleDbRequests().addNewEnv(env);
     manageDatabase.loadEnvMapForOneTenant(tenantId);
```

Comment:
may be we should also add loadClustersForOneTenant ?

Reply:
I think a new entry method for entry would be better to prevent us from pulling all the data from the db and reloading it everytime